src/accelerate/utils/operations.py

```diff
@@ -298,6 +298,13 @@ def _gpu_gather_one(tensor):
         if not tensor.is_contiguous():
             tensor = tensor.contiguous()
 
+        # Check if `tensor` is not on CUDA
+        if state.device.type == "cuda" and tensor.device.type != "cuda":
```

Comment:
Are there other device mismatches that could be checked here?

Reply:
@muellerzr  @BenjaminBossan  Can this logic be extended to other devices ?  Seems like a generic exception handling case.

Reply:
Yes it can, for now it’s just a thing on CUDA but if it’s useful for other devices that can be added. This is just a known base case